src/test/java/org/wildfly/bot/utils/testing/internal/TestModel.java

```diff
@@ -0,0 +1,138 @@
+package org.wildfly.bot.utils.testing.internal;
+
+import io.quarkiverse.githubapp.testing.dsl.GitHubMockSetup;
+import io.quarkus.test.junit.callback.QuarkusTestAfterAllCallback;
+import io.quarkus.test.junit.callback.QuarkusTestBeforeEachCallback;
+import io.quarkus.test.junit.callback.QuarkusTestContext;
+import io.quarkus.test.junit.callback.QuarkusTestMethodContext;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.graphql.NonNull;
+import org.wildfly.bot.utils.PullRequestJsonBuildable;
+import org.wildfly.bot.utils.testing.CheckedFunction;
+import org.wildfly.bot.utils.testing.PullRequestJson;
+import org.wildfly.bot.utils.testing.PullRequestJsonBuilder;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+public class TestModel implements QuarkusTestBeforeEachCallback, QuarkusTestAfterAllCallback {
+
+    private static final String POLLING_PROFILE = "polling";
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private static Optional<Callable<? extends PullRequestJsonBuilder>> pullRequestJsonBuilderCallableSse;
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private static Optional<Callable<? extends PullRequestJsonBuilder>> pullRequestJsonBuilderCallablePooling;
+    private static CheckedFunction<PullRequestJsonBuilder, ? extends PullRequestJsonBuilder> pullRequestJsonBuilderFunction = builder -> builder;
+    private static PullRequestJson pullRequestJson;
+
+    /**
+     * @implNote Between tests we clear generated Json, corresponding Building
+     *           from {@code pullRequestJsonBuilderFunction} and triggers
+     */
+    @Override
+    public void beforeEach(QuarkusTestMethodContext context) {
+        pullRequestJson = null;
+        TestModel.pullRequestJsonBuilderFunction = builder -> builder;
```

Comment:
Yes, i.e. we don't create any `PullRequestJsonBuildable` in tests, which would manipulate the json. So instead of checking, whether the user has provided some builder, we instead return identity.

Example `.setPullRequestJsonBuilderBuild(builder -> builder.title("Hi"))` would translate to something like `TestModel.pullRequestJsonBuilderFunction = builder -> builder.title("Hi");`
So we can then always call `builder.apply()`;

Reply:
I mean there should be `Function.identity()`

Reply:
There is a problem, that it's not an identity in reality. As first param is `PullRequestJsonBuilder` and second is `? extends PullRequestJsonBuilder`, which causes the problem. But thanks for pointing me to `Function.identity()`, didn't know it.